compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BNilType.java

```diff
@@ -48,4 +48,9 @@ public String getDesc() {
     public String toString() {
         return Names.NIL_VALUE.value;
     }
+
+    @Override
+    public boolean hasImplicitInitialValue() {
```

Comment:
can we move this to the BType class as the default implementation?
Then anyone can override when needed..

Reply:
+1.